src/plugins/intel_gpu/src/graph/deconvolution.cpp

```diff
@@ -134,7 +134,7 @@ std::vector<layout> deconvolution_inst::calc_output_layouts(deconvolution_node c
         out_fmt = node.get_preferred_output_fmt();
     }
 
-    if (!node.get_program().is_new_shape_infer() && desc->with_output_size) {
+    if (desc->with_output_size) {
```

Comment:
I think that for the new shape infer it would be better to remake legacy `tensor` to `ShapeType`
And it looks like this code could never have been called before, since `calc_output_layouts` is only called with new shape infer